pkg/ebpf/events_pipeline.go

```diff
@@ -233,40 +231,47 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
-				ParentProcessID:     int(ctx.Ppid),
-				HostProcessID:       int(ctx.HostPid),
-				HostThreadID:        int(ctx.HostTid),
-				HostParentProcessID: int(ctx.HostPpid),
-				UserID:              int(ctx.Uid),
-				MountNS:             int(ctx.MntID),
-				PIDNS:               int(ctx.PidID),
-				ProcessName:         string(bytes.TrimRight(ctx.Comm[:], "\x00")),
-				HostName:            string(bytes.TrimRight(ctx.UtsName[:], "\x00")),
-				CgroupID:            uint(ctx.CgroupID),
-				ContainerID:         containerData.ID,
-				Container:           containerData,
-				Kubernetes:          kubernetesData,
-				EventID:             int(ctx.EventID),
-				EventName:           eventDefinition.Name,
-				MatchedPolicies:     ctx.MatchedPolicies,
-				ArgsNum:             int(ctx.Argnum),
-				ReturnValue:         int(ctx.Retval),
-				Args:                args,
-				StackAddresses:      stackAddresses,
-				ContextFlags:        flags,
-				Syscall:             syscall,
+				Timestamp:             int(ctx.Ts),
+				ThreadStartTime:       int(ctx.StartTime),
+				ProcessorID:           int(ctx.ProcessorId),
+				ProcessID:             int(ctx.Pid),
+				ThreadID:              int(ctx.Tid),
+				ParentProcessID:       int(ctx.Ppid),
+				HostProcessID:         int(ctx.HostPid),
+				HostThreadID:          int(ctx.HostTid),
+				HostParentProcessID:   int(ctx.HostPpid),
+				UserID:                int(ctx.Uid),
+				MountNS:               int(ctx.MntID),
+				PIDNS:                 int(ctx.PidID),
+				ProcessName:           string(bytes.TrimRight(ctx.Comm[:], "\x00")),
+				HostName:              string(bytes.TrimRight(ctx.UtsName[:], "\x00")),
+				CgroupID:              uint(ctx.CgroupID),
+				ContainerID:           containerData.ID,
+				Container:             containerData,
+				Kubernetes:            kubernetesData,
+				EventID:               int(ctx.EventID),
+				EventName:             eventDefinition.Name,
+				MatchedPoliciesKernel: ctx.MatchedPolicies,
+				ArgsNum:               int(ctx.Argnum),
+				ReturnValue:           int(ctx.Retval),
+				Args:                  args,
+				StackAddresses:        stackAddresses,
+				ContextFlags:          flags,
+				Syscall:               syscall,
 			}
 
-			if !t.shouldProcessEvent(&evt) {
-				_ = t.stats.EventsFiltered.Increment()
-				// we can stop processing the event if it doesn't have derivatives.
-				// otherwise, since it has MatchedPolicies==0 it won't be emitted later anyway.
+			// If there aren't any policies that need filtering in userland, tracee **may** skip
+			// this event, as long as there aren't any derivatives that depend on it. Some base
+			// events (for derivative ones) might not have set related policy bit, thus the need
+			// to continue with those within the pipeline.
+			//
+			// TODO: should the base event set policy bit for policies that depend on their
+			//       derivative events ? (check matchPolicies)
+			//
+
+			if t.matchPolicies(&evt) == 0 {
```

Comment:
This logic will break signatures. Because just like derivatives the signatures are not the events coming from kernel, they expect a base event to be parsed in a later stage. For `anti_debugging` for example, we need to allow ptrace to pass because it is a base event, but ptrace itself wasn't matched in any policy.

Reply:
updating the branch worked, the bug was in a previous version

Reply:
but let's update the comment above to mention signatures too